Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/transformation/TransformationSaveService.java

```diff
@@ -0,0 +1,92 @@
+package edu.asu.spring.quadriga.service.impl.transformation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import edu.asu.spring.quadriga.domain.workspace.ITransformationFile;
+import edu.asu.spring.quadriga.exceptions.FileStorageException;
+import edu.asu.spring.quadriga.service.transformation.ITransformationSaveService;
+import edu.asu.spring.quadriga.utilities.IFileSaveUtility;
+
+/**
+ * This service contains method to save contents of transformation files to
+ * local file storage
+ * 
+ * @author yoganandakishore
+ *
+ */
+
+@Service
+public class TransformationSaveService implements ITransformationSaveService {
+
+    @Qualifier("transformfileSaveUtil")
+    @Autowired
+    private IFileSaveUtility transformationFileManager;
+
+    @Override
+    public boolean saveFileToLocal(ITransformationFile transformationFile) throws FileStorageException {
+
+        String directoryName = getTransformationDirectoryName(transformationFile);
+        String metaDataContent = getMetaDataContent(transformationFile);
+
+        return (transformationFileManager.saveFiletoDir(directoryName, "metadata.txt", metaDataContent.getBytes())
+                && transformationFileManager.saveFiletoDir(directoryName,
+                        "Pattern_" + transformationFile.getPatternFileName(),
+                        transformationFile.getPatternFileContent())
+                && transformationFileManager.saveFiletoDir(directoryName,
+                        "Mapping_" + transformationFile.getMappingFileName(),
+                        transformationFile.getMappingFileContent()));
+    }
+
+    /**
+     * forms the directory name by appending username followed by
+     * "Transformation" followed by mapping file name and transformation file
+     * names, each separated by underscores.
+     * 
+     * @param transformationFile
+     * @return String which contains the directory name
+     */
+    private String getTransformationDirectoryName(ITransformationFile transformationFile) {
```

Comment:
you directory name should be the id of the transformation. this way you don't need to do any string building when retrieving content. Just get the id of the transformation file object in the db and use that to retrieve files. It will also ensure the directory name is always unique.